geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/expression/FastPropertyIsEqualTo.scala

```diff
@@ -23,6 +23,18 @@ abstract class FastPropertyIsEqualTo(exp1: Expression, exp2: Literal) extends Pr
   override def getMatchAction: MatchAction = MatchAction.ANY
 
   override def toString: String = s"[ $exp1 = $exp2 ]"
+
+  def canEqual(other: Any): Boolean = other.isInstanceOf[FastPropertyIsEqualTo]
+
+  override def equals(other: Any): Boolean = other match {
+    case that: FastPropertyIsEqualTo =>
+      (that canEqual this) && exp1 == that.getExpression1 && exp2 == that.getExpression2 &&
```

Comment:
Should we call exp1 == ... or exp1.equals(...)?  (Me just asking silly questions that I cannot remember the answer to.)

Reply:
`==` in scala is equivalent to java `.equals`, although it's also null safe